src/HOI4World/Map/Railways.cpp

```diff
@@ -228,30 +228,52 @@ std::vector<std::shared_ptr<Vic2::Province>> GetVic2PathProvincesFromNumbers(con
 // 5|0 1 1 2 2 3 3
 // 6|0 1 2 2 3 3 3
 int GetRailwayLevel(const std::vector<int>& vic2_possible_path,
-	 const std::map<int, std::shared_ptr<Vic2::Province>>& vic2_provinces)
+	 const std::map<int, std::shared_ptr<Vic2::Province>>& vic2_provinces,
+	 const Mappers::ProvinceMapper& province_mapper,
+	 const std::map<int, HoI4::Province>& hoi4_provinces,
+	 const HoI4::States& hoi4_states)
 {
 	const auto vic2_path_provinces = GetVic2PathProvincesFromNumbers(vic2_possible_path, vic2_provinces);
 	if (vic2_path_provinces.size() != vic2_possible_path.size())
 	{
 		return 0;
 	}
 
-	std::vector<int> rail_levels;
+	std::vector<double> rail_levels;
+	bool capitalRailway = false;
 	for (const auto& vic2_province: vic2_path_provinces)
 	{
-		const auto rail_level = vic2_province->getRailLevel();
+		auto number = vic2_province->getNumber();
+		auto HoI4ProvNumber = province_mapper.getVic2ToHoI4ProvinceMapping(number);
+		auto HoI4Province = hoi4_provinces.find(HoI4ProvNumber[0]);
+		auto terrainType = HoI4Province->second.getType();
+		auto rail_level = static_cast<double>(vic2_province->getRailLevel()) / 3.0;
 		if (rail_level == 0)
-		{
 			return 0;
+		if (terrainType == "mountain" || terrainType == "jungle" || terrainType == "desert")
+			rail_level -= 2;
+		else if (terrainType == "hills" || terrainType == "marsh")
+			rail_level--;
+		else // forest, urban, plains
+			rail_level++;
```

Comment:
Use ```{``` and ```}``` even for single-line expressions after conditionals

Reply:
Is there a style guide for the converter projects? Dont want to cause you unnecessary pull request headaches due to style differences.
Assuming curly brackets here are for easier maintainability?

Reply:
Nothing formally written up, sadly. And Zemurin and I have different stylistic preferences between converters.

Mine is mostly the Google style guide, but with longer line widths and opening braces on the following line. For Vic3 to HoI4, I'm hoping to get clang-tidy to enforce it all. It's probably too late for this converter.